src/Gaffer/ValuePlug.cpp

```diff
@@ -239,11 +239,17 @@ class ValuePlug::HashProcess : public Process
 			const Context *currentContext = threadState.getContext();
 			const HashProcessKey processKey( p, plug, currentContext, p->m_dirtyCount, computeNode, computeNode ? computeNode->hashCachePolicy( p ) : CachePolicy::Uncached );
 
-			if( processKey.cachePolicy == CachePolicy::Uncached || threadState.disallowCacheRead( plug, ThreadState::CacheType::Hash ) )
+			if( processKey.cachePolicy == CachePolicy::Uncached )
 			{
 				HashProcess process( processKey );
 				return process.m_result;
 			}
+			else if( threadState.disallowCacheRead( plug, ThreadState::CacheType::Hash ) )
+			{
+				HashProcess process( processKey );
+				g_globalCache.set( processKey, process.m_result, 1 );
```

Comment:
Depending on the cache policy, this might not be the right cache to set it in. We only ever perform lookups in the global cache for the `TaskCollaboration` and `TaskIsolation` policies, and we rely on very low eviction rates from the global cache. So for policies that use the local cache alone, we should set into the local cache.